source/mainnet/smart-contracts/tutorials/piggy-bank/deploying.rst

```diff
@@ -0,0 +1,160 @@
+
+If everything works correctly the output has a green line with your module reference. You can find additional information about deploying a module in this :ref:`guide<deploy-module>`.
+
+.. code-block:: console
+
+   Module successfully deployed with reference: <module-reference>.
+   Module reference <module-reference> was successfully named 'piggy_bank_part2_module'.
+
+
+Initializing a module
+=====================
+
+You can initialize your smart contract now to create a smart contract instance. Let's give the instance the name ``piggy_bank_part2_instance`` so it can be referenced in this tutorial.
+
+.. code-block:: console
+
+   $./concordium-client contract init piggy_bank_part2_module --sender <account-name> --contract PiggyBank --name piggy_bank_part2_instance --energy 1000 --grpc-port 10001
+
+.. Note::
+   1000 is enough energy to initialize the piggy bank smart contract but you might need more energy for larger smart contracts.
+
+
+If everything works correctly the output has a green line with your smart contract instance index.
+
+.. code-block:: console
+
+   Contract successfully initialized with address: {"index":<smart-contract-instance-index>,"subindex":0}
+   Contract address {"index":<smart-contract-instance-index>,"subindex":0} was successfully named 'piggy_bank_part2_instance'.
+
+
+You can find additional information about initializing a smart contract instance in this :ref:`guide<initialize-contract>`.
+
+
+Updating the piggy bank smart contract (sending CCD)
+====================================================
+
+Let's deposit 1 CCD into the piggy bank to test if you can update the smart contract instance.
+
+.. code-block:: console
+
+   $./concordium-client contract update piggy_bank_part2_instance --entrypoint insert --energy 1000 --sender <account-name> --amount 1 --grpc-port 10001
+
+If everything works correctly the output has a green line as follows:
+
+.. code-block:: console
+
+   Successfully updated contract instance {"index":<smart-contract-instance-index>,"subindex":0} ('piggy_bank_part2_instance') using the function 'insert'.
+
+You can find additional information about updating a smart contract instance in this :ref:`guide<interact-instance>`.
+
+
+Viewing the piggy bank state
+============================
+
+You can display the raw return value of your piggy bank smart contract with the following command.
+
+.. code-block:: console
+
+   $./concordium-client contract invoke piggy_bank_part2_instance --entrypoint view --energy 1000 --grpc-port 10001
+
+If everything works correctly the output has a green line as follows:
+
+.. code-block:: console
+
+   Invocation resulted in success:
+   - Energy used: 510 NRG
+   - Return value (raw):
+   [0,64,66,15,0,0,0,0,0]
+   .
+
+
+You can find additional information about invoking in this :ref:`guide<invoke-instance>`.
+
+
+Updating the piggy bank smart contract (sending no CCD)
```

Comment:
With the comment from my previous review, I didn't mean that this section should have such a focus on the `no CCD` part. I thought of it as a side note that would illustrate the difference between calling payable and non-payable entrypoints. I thought the primary focus of this section would be to show the smash entrypoint, which is the only entrypoint not already covered.

But anyhow, seeing it now, I don't think this section adds so much value to the tutorial. You can remove if it you want, or, alternatively, change the title to be smashing focused rather than `sending no CCD`.

Reply:
I think the second option is better. With an explicit point about payable/non-payable and what the response would be if they mixed it up.